### ISSUE TITLE
Error validating nullable readonly properties with null value

### ISSUE THREAD

**Issue by tjasek225** [luboshl/MiniValidationPlus#16]
When validating nullable readonly property IsNonNullableType value is true but should be false. It results in property being invalid with validation result "The XXX field is required."

```csharp
public record TestClass
{
    [MaxLength(20)]
    public string? Text { get; } // Validating instance of this class with Text equal to null will result in error described above.

    public TestClass(string? text)
    {
        Text = text;
    }
}
```

**Comment by luboshl** [comment]
Hi @tjasek225, thank you for reporting the issue. I've investigated that and I can reproduce the problem. The problem is for a property with nullable reference type **without** annotation. So when I extend your sample, the problem is with the second property:

```csharp
public record TestClass
{
    [MaxLength(20)]
    public string? Text { get; } // This property is ok

    public string? Text2 { get; } // Validation of this property is failing

    public TestClass(string? text, string? text2)
    {
        Text = text;
        Text2 = text2;
    }
}
```
@tjasek225, can you check your real scenario is for nullable property without annotation?

I've created failing tests to repro the issue here: https://github.com/luboshl/MiniValidationPlus/commit/b6ba1c0e55743304448e6e3522a89c0afca2e445

Tests that fail:
- `NonNullablePropertyHelperTests.IsNonNullableReferenceType_Identifies_Correct_Properties_Of_Class`
- `TryValidate.PropertyWithoutSetter_Nullable_Valid_When_Null`

**Comment by luboshl** [comment]
@tjasek225, bug was fixed in #18 and released in [v1.2.2](https://github.com/luboshl/MiniValidationPlus/releases/tag/v1.2.2).
